src/main/java/com/openlattice/authorization/HazelcastAuthorizationService.java

```diff
@@ -83,6 +84,39 @@ public void setSecurableObjectType( AclKey aclKey, SecurableObjectType objectTyp
         aces.executeOnEntries( new SecurableObjectTypeUpdater( objectType ), hasAclKey( aclKey ) );
     }
 
+    private void signalMaterializationPermissionChange(
+            AclKey key,
+            Principal principal,
+            EnumSet<Permission> permissions,
+            SecurableObjectType securableObjectType
+    ) {
+        // if there is a change in materialization permission for a property type for an organization principal,
+        // we need to flag it
+        if ( permissions.contains( Permission.MATERIALIZE )
+                && principal.getType().equals( PrincipalType.ORGANIZATION )
+                && securableObjectType.equals( SecurableObjectType.PropertyTypeInEntitySet ) ) {
+            eventBus.post( new MaterializePermissionChangeEvent( principal, key.get( 0 ) ) );
+        }
+    }
+
+    private void signalMaterializationPermissionChangeBulk(
+            Set<AceKey> aceKeys,
+            EnumSet<Permission> permissions,
+            SecurableObjectType securableObjectType
+    ) {
+        // if there is a change in materialization permission for a property type for an organization principal,
+        // we need to flag it
+        if ( permissions.contains( Permission.MATERIALIZE )
```

Comment:
Do you know if this is normally called on a huge number of aceKeys at once? It would be nice for this to just call through to `signalMaterializationPermissionChange` instead of duplicating the logic here

Reply:
I'm not sure, there are not metrics yet on `PermissionsController.updateAcls`.

Reply:
This should only get called for # property types x # of organizations changed, which should generally be one. I think Katherine's comment is valid and we should consider not firing off a bunch of events, especially if they can bulked more effectively (each event blocks a thread for the duration of processing).

Reply:
I think Katherine is implying, that we should reuse the code if this call is used mostly with single (or just few) permission updates. And I guess, you would change the call rather to go through the `updates` (`SetMultimap<AceValue, AceKey>`) one more time and only fire 1 event per 1 organization principal, right?